Modules/mathmodule.c

```diff
@@ -2016,6 +2016,62 @@ math_factorial(PyObject *module, PyObject *arg)
 }
 
 
+/*[clinic input]
+math.lcm
+
+    x as a: object
+    y as b: object
+    /
+least common multiple of x and y
+[clinic start generated code]*/
+
+static PyObject *
+math_lcm_impl(PyObject *module, PyObject *a, PyObject *b)
+/*[clinic end generated code: output=6f83fb6d671074ba input=bd41b785dc2a4ff1]*/
+{
+    PyObject *g, *m, *f, *ab;
+    
+    a = PyNumber_Index(a);
+    if (a == NULL) {
+        return NULL;
+    }
+    b = PyNumber_Index(b);
+    if (b == NULL) {
+        Py_DECREF(a);
+        return NULL;
+    }
+    if (_PyLong_Sign(a) == 0 && _PyLong_Sign(b) == 0) {
```

Comment:
One tiny optimization: there's no point in checking `b` here, since if `a` is zero then the lcm is zero regardless of what `b` is. So I'd suggest just dropping the second check here.

Alternatively (and probably better), you could change the `&&` to a `||` so that we do an early return if either `a` or `b` is zero.